modules/renter/hostdb/online.go

```diff
@@ -0,0 +1,39 @@
+package hostdb
+
+import (
+	"time"
+)
+
+// online.go performs regular checks against the gateway to see if there is
+// internet connectivity. The check is performed by looking for at least one
+// non-local peer in the peer list. If there is no internet connectivity, scans
+// must be stopped lest we penalize otherwise online hosts.
+
+func (hdb *HostDB) threadedOnlineCheck() {
```

Comment:
I searched around and it seems like the only robust alternative to this is to ping a high-uptime server like `8.8.8.8`. Obviously there are disadvantages to that, but at least we wouldn't need to thread a new dependency through all this code. It could also be used as a fallback, though it's probably a bad sign if you don't have any peers.